x/accounts/genesis.go

```diff
@@ -64,13 +64,14 @@ func (k Keeper) exportAccount(ctx context.Context, addr []byte, accType string,
 }
 
 func (k Keeper) ImportState(ctx context.Context, genState *v1.GenesisState) error {
-	err := k.AccountNumber.Set(ctx, genState.AccountNumber)
-	if err != nil {
-		return err
-	}
-
 	// import accounts
 	for _, acc := range genState.Accounts {
+		// increase the account number
+		_, err := k.AccountNumber.Next(ctx)
+		if err != nil {
+			return fmt.Errorf("%w: %s", err, acc.Address)
+		}
```

Comment:
what is the reason we are increasing here? seems like the number can just be exported then imported as normal no?

Reply:
agreed, wondering the same

Reply:
This code can easily create duplicate account numbers when the genesis was manually modified and accounts removed.... 💣

Reply:
so i have 2 problems which make the tests falling, the setting of genesis account number value called during upgrade handler which set the value to default ( in this case 0 ) which broke the auth accounts index --> failed the system test. And the import of export state test seems to fail due to DiffKVStores in the accounts store, which i think is due to the account number value since there are no accounts in the genesis file. And also if the account number is set directly like this can cause a logic flaw where for example auth has 10 accounts so acc num is 10 when export and import again the acc num is 10 in the beginning, so when auth init genesis, for 10 accounts, acc num need increment to get the value to set to the account which then make acc num becomes 20 . So i need a way to set the account number indirectly hence this logic

Reply:
hmmmm we can set the value of the account number to the highest account number in the genesis state so it won't create duplicate

Reply:
yes, that is a feasible way. Not sure if it has to be +1 though

Reply:
updated the logic